### PR TITLE
feat(Interaction): support moving knobs

### PR DESCRIPTION
**Pull request by 10FingerArmy**
Knobs have been converted to use configurable joints. This enables
specifying connected rigidbodies like in many other 3d controls already.
Scene 25 contains a new knob on the small door now showcasing this.

While at it also the axis has been switch from world to local so
existing knobs will have to be adapted.

Also the value calculation was fixed which was working more by accident
than by intention. It now uses a heuristics to determine the correct
axis.

### CODE REVIEW

File: Assets/VRTK/Scripts/Controls/3D/VRTK_Knob.cs [hunk 1]
```diff
@@ -163,8 +196,23 @@ private KnobDirection DetectDirection()
 
         private float CalculateValue()
         {
+            if (!subDirectionFound)
```
Comment (pvh) [inline Assets/VRTK/Scripts/Controls/3D/VRTK_Knob.cs]: should be able to just use subDirection and drop subDirectionFound, right?

File: Assets/VRTK/Scripts/Controls/3D/VRTK_Knob.cs [hunk 2]
```diff
@@ -163,8 +196,23 @@ private KnobDirection DetectDirection()
...
+            {
+                float angleX = Mathf.Abs(transform.localRotation.eulerAngles.x - initialLocalRotation.x) % 90;
+                float angleY = Mathf.Abs(transform.localRotation.eulerAngles.y - initialLocalRotation.y) % 90;
+                float angleZ = Mathf.Abs(transform.localRotation.eulerAngles.z - initialLocalRotation.z) % 90;
+                angleX = (Mathf.RoundToInt(angleX) > 89 && Mathf.RoundToInt(angleX) < 91) ? 0 : angleX;
```
Comment (pvh) [inline Assets/VRTK/Scripts/Controls/3D/VRTK_Knob.cs]: What does this do? You're already % 90 above so shouldn't it always be 0 <= angle <= 90?

File: Assets/VRTK/Scripts/Controls/3D/VRTK_Knob.cs [hunk 3]
```diff
@@ -163,8 +196,23 @@ private KnobDirection DetectDirection()
...
+                    subDirectionFound = true;
+                }
+            }
+
             float angle = 0;
-            switch (finalDirection)
+            switch (subDirection)
             {
                 case KnobDirection.x:
                     angle = transform.localRotation.eulerAngles.x - initialLocalRotation.x;
```
Comment (pvh) [inline Assets/VRTK/Scripts/Controls/3D/VRTK_Knob.cs]: A few lines below (GitHub won't let me comment outside the patch) you round * 1000 then divide back by 1000. Math.Round has a two argument form where you can provide a precision. Math.Round(angle, 3), for example.

File: Assets/VRTK/Scripts/Controls/3D/VRTK_Knob.cs (hunk 2 above)
Comment (10FingerArmy) [inline Assets/VRTK/Scripts/Controls/3D/VRTK_Knob.cs]: Good catch I'll try to simplify that

File: Assets/VRTK/Scripts/Controls/3D/VRTK_Knob.cs (hunk 3 above)
Comment (10FingerArmy) [inline Assets/VRTK/Scripts/Controls/3D/VRTK_Knob.cs]: Math.Round only supports doubles, Unity works with floats, that's why there is the Mathf class, which unfortunately does not have the necessary rounding to digits functionality. Would you judge it's better to cross-cast to use the other method?

**Comment by 10FingerArmy** [comment]
Should be ready now for merge

File: Assets/VRTK/Scripts/Controls/3D/VRTK_Knob.cs (hunk 3 above)
Comment (pvh) [inline Assets/VRTK/Scripts/Controls/3D/VRTK_Knob.cs]: Nah, this is janky but fine as long as it's always done the same.

File: Assets/VRTK/Scripts/Controls/3D/VRTK_Knob.cs (hunk 1 above)
Comment (pvh) [inline Assets/VRTK/Scripts/Controls/3D/VRTK_Knob.cs]: Gotcha - C# is not my main language.

**Pull request merged**